packages/forgae-utils/utils/forgae-utils.js

```diff
@@ -139,9 +146,70 @@ function readSpawnOutput (spawnResult) {
     return buffMessage.toString('utf8');
 }
 
-async function contractCompile (source, options) {
-    const compiler = await ContractCompilerAPI(options);
-    return compiler.compileContractAPI(source);
+async function contractCompile (source, contractPath, compileOptions) {
+    let result;
+    let options = {
+        "file_system": null
+    }
+    dependencies = {};
+
+    getDependencies(source, contractPath)
+
+    options["file_system"] = dependencies
+
+    let body = {
+        code: source,
+        options
+    };
+    
+    result = await axios.post(compileOptions.compilerUrl, body, options);
+
+    return result;
+}
+
+function getDependencies (contractContent, contractPath) {
+    let allDependencies = [];
+    let dependencyFromContract;
+    let dependencyContractContent;
+    let dependencyContractPath;
+    let actualContract;
+
+    match = rgx.exec(contractContent)
+
+    if (match == null) {
+        return;
+    }
+
+    allDependencies = contractContent.match(rgx)
+    try {
+        for (let index = 0; index < allDependencies.length; index++) {
+            dependencyFromContract = dependencyPathrgx.exec(allDependencies[index])
+            dependencyPathrgx.lastIndex = 0;
+
+            contractPath = mainContractsPathRgx.exec(contractPath)
+            mainContractsPathRgx.lastIndex = 0;
+            dependencyContractPath = path.resolve(`${ contractPath[0] }/${ dependencyFromContract[1] }`)
+            dependencyContractContent = fs.readFileSync(dependencyContractPath, 'utf-8')
+            actualContract = getActualContract(dependencyContractContent)
+
+            if (!dependencies[dependencyFromContract[1]]) {
+                dependencies[dependencyFromContract[1]] = actualContract;
+            }
+
+            getDependencies(dependencyContractContent, dependencyContractPath)
+        }
+
+    } catch (e) {
```

Comment:
nobody will understand that there is something wrong, it throws silently. you need to show it